### PR TITLE
kb(carousel):thumbnail scrollable navigation

### PR DESCRIPTION
**Pull request by xristianstefanov**
- Added Kb for implementing Carousel Thumbnail Navigation.

### CODE REVIEW

File: knowledge-base/carousel-thumbnail-scrollable-navigation.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
+---
+title: Carousel Thumbnail Scrollable navigation
```
Comment (dimodi) [inline knowledge-base/carousel-thumbnail-scrollable-navigation.md:2]:
```suggestion
title: Carousel Thumbnail Scrollable Navigation
```

File: knowledge-base/carousel-thumbnail-scrollable-navigation.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+description: How to create Thumbnail Scrollable Navigation in Carousel?
```
Comment (dimodi) [inline knowledge-base/carousel-thumbnail-scrollable-navigation.md:3]:
```suggestion
description: How to use thumbnails in scrollable navigation for the Carousel?
```

File: knowledge-base/carousel-thumbnail-scrollable-navigation.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+type: how-to
+page_title: Carousel Thumbnail Scrollable navigation
```
Comment (dimodi) [inline knowledge-base/carousel-thumbnail-scrollable-navigation.md:5]:
```suggestion
page_title: Carousel Thumbnail Scrollable Navigation
```

File: knowledge-base/carousel-thumbnail-scrollable-navigation.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+## Description
+How to add a thumbnail scrollable navigation below the page/image in Carousel? This would be a nice addition, which would allow independent scrolling from that of the current page. It will also be clickable in the same way that the dots are to jump to the selected page.
+
+## Solution
+To add a thumbnail scrollable navigation:
+
+1. Add HTML `<div>` container under the `Carousel` markup.
+2. Apply custom CSS with the needed styles to the container.
+3. Inside the `<div>`, loop through all the images in `Carousel` and define them in a smaller size.
+4. Optionally, use a javascript function that keeps the synchronization of the `AutomaticPageChange` and the thumbnail.
```
Comment (dimodi) [inline knowledge-base/carousel-thumbnail-scrollable-navigation.md:33]:
```suggestion
1. Render the thumbnails in a `<div>` container under the `Carousel`.
1. Use a `click` handler for the thumbnails to page the Carousel programmatically.
1. Optionally, use JavaScript to synchronize the scroll position of the thumbnails with the current Carousel page. This applies to the case when the user pages the Carousel with the built-in component buttons.
```

**Review by dimodi — APPROVED** [review]
Updated the example code to:
* include the JS code in the Razor file.
* demonstrate thumbnail scrollability - when you implement a nice hack, make it show

**Comment by marin-bratanov** [comment]
Add a note that the js function belongs to a proper place and the supress-error hack is a hack for development. Consider commenting out the console.log calls in it.

File: knowledge-base/carousel-thumbnail-scrollable-navigation.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+                 src="@img.ImageUrl" alt="Photograph" />
+        }
+    </div>
+</div>
+
+<script suppress-error="BL9992">
+    function ScrollToCurrentPage(imgId) {
+        var elem = document.getElementById(imgId);
+        console.log(imgId);
+        console.log(elem);
```
Comment (dimodi) [inline knowledge-base/carousel-thumbnail-scrollable-navigation.md:68]:
```suggestion
```

File: knowledge-base/carousel-thumbnail-scrollable-navigation.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        @foreach (var img in CarouselData)
+        {
+            <img @onclick="@(() => PageChangedHandler(img.ImageID))" id="@img.ImageID"
+                 class="image-thumbnail"
+                 src="@img.ImageUrl" alt="Photograph" />
+        }
+    </div>
+</div>
+
+<script suppress-error="BL9992">
```
Comment (dimodi) [inline knowledge-base/carousel-thumbnail-scrollable-navigation.md:64]:
```suggestion
<!-- Move the JavaScript code to a separate .js file in the actual app -->
<script suppress-error="BL9992">
```

**Pull request merged**